### ISSUE TITLE
Imported RKE Cluster displays %3Cnil%3E for NodePorts in Rancher Server UI

### ISSUE THREAD

**Issue by azbpa** [rancher/rancher#19208]
Facing a similar issue described here rancher/rancher#15482 when importing a cluster created manually with RKE `v0.1.16` to Rancher Server `v2.1.6`.

Not sure if it is a bug or maybe a misconfiguration. 

From where comes the information that is used by Rancher UI to display the endpoint of the host? Is it a custom annotation that might be missing? Or is it based on an option to be set during RKE installation in `cluster.yml`?

**Comment by davidnuzik** [comment]
Testing note: Tested in v2.1.7 and cannot reproduce.

**Comment by azbpa** [comment]
Can you share your `cluster.yml` so I could identify some differences to mine?

Further I noticed on the node section in Rancher UI that I can see both the IP addresses and the FQDNs of my RKE node(s):

`node01.domain.local  /  10.20.30.40`

When I deploy a custom cluster from Rancher UI I only see the IP addresses of the nodes. 

Might this be relevant?

**Comment by davidnuzik** [comment]
![Screen Shot 2019-04-02 at 1 23 43 PM](https://user-images.githubusercontent.com/45179589/55433671-ab379c00-554a-11e9-9130-a57426b974e6.png)


I just set up a simple 1 node cluster.yml with aws ec2 instance (t2.medium in my case)
```
nodes:
- address: secret_IP
  user: ubuntu
  role: [controlplane,worker,etcd]
  internal_address: 172.99.99.99
```

What does your cluster.yml look like for your rke cluster? (Hide IP/DNS/sensitive info please)

**Comment by davidnuzik** [comment]
And my workload:
![image](https://user-images.githubusercontent.com/45179589/55433790-f3ef5500-554a-11e9-8295-4db7a9370f21.png)

**Comment by azbpa** [comment]
Played around a bit. Issue seems to be related to what I am setting as `address` in `cluster.yml`

With IP address it is working (like in your example): `http://10.20.30.40:31587/`
```
cluster_name: mycluster
nodes:
- address: 10.20.30.40
  user: rke
  role: [controlplane,worker,etcd]
```

However, when using FQDN `mynode01.domain.local` instead, Rancher UI shows the wrong url like: `http://%3Cnil%3E:31587/`
```
cluster_name: mycluster
nodes:
- address: mynode01.domain.local
  user: rke
  role: [controlplane,worker,etcd]
```

So, technically my RKE cluster is working with both approaches. However I would prefer to use the FQDN since the IP might be volatile.

I think, Rancher UI should be able to handle both IP addresses and FQDNs.

**Comment by davidnuzik** [comment]
Thanks @azbpa I'll give it another look using FQDN this time.

**Comment by tfiduccia** [comment]
Version - master 4/11.  
URL encoding needs to be fixed.

**Comment by azbpa** [comment]
just as update: I still have this behavior in Rancher v2.2.2

**Comment by vincent99** [comment]
@alena1108 I think the address is being sent wrong in the endpoints, `<nil>` is clearly from Go and not the UI (where we have `null` and `undefined`, but not `nil` ).

**Comment by davidnuzik** [comment]
For testing: 
Reference comment https://github.com/rancher/rancher/issues/19208#issuecomment-480225647
Notice that to reproduce we need to have our node use a DNS name rather than static ip. You can do this via AWS - each instance should have a long domain name but this should still work to repro and validate.

**Comment by izaac** [comment]
Rancher version: master-head (v2.3) (08/21/2019)

I reproduced the issue in Rancher 2.2.2:

<img width="619" alt="Screen Shot 2019-08-21 at 4 19 07 PM" src="https://user-images.githubusercontent.com/27659/63474880-955e4100-c42f-11e9-958e-d1a76eac9646.png">


Validated the fix in v2.3.

<img width="704" alt="Screen Shot 2019-08-21 at 4 20 54 PM" src="https://user-images.githubusercontent.com/27659/63474921-bc1c7780-c42f-11e9-8f52-947401a93309.png">

I've used a temporary Instance, so the information in the screenshot won't be valid anymore. As I had to put the full FQDN to demonstrate the fix.
